### PR TITLE
Parser support library

### LINKED ISSUE #174: Support the Library call
This valid brightscript causes parse errors: 
```
Library "v30/bslCore.brs"
```

image

As a first step, could you update the parser to support this syntax? You can implement the actual functionality in the interpreter later, but it would be useful for parser consumers to not get syntax errors.

Here's the documentation for `library`. https://sdkdocs.roku.com/display/sdkdoc/Component+Architecture#ComponentArchitecture-ScriptLibraries

### PR DESCRIPTION
**Pull request by TwitchBronBron**
This PR provides parser support for the `library` keyword. Runtime support will need to be implemented at a later time (tracked by #217)
Fixes #174

### CODE REVIEW

**Comment by TwitchBronBron** [comment]
@sjbarag I just got this branch up-to-date from master

File: src/parser/Parser.ts [hunk 1]
```diff
@@ -200,6 +200,13 @@ export class Parser {
         let current = 0;
         let tokens = toParse;
 
+        //the depth of the calls to function declarations. Helps some checks know if they are at the root or not.
```
Comment (sjbarag) [inline src/parser/Parser.ts]: I played around with `Library` statements to confirm something I think I heard but can't find an authoritative source for: that `Library` statements must only be at the top of the file, and can't occur after any non-library statements (including named function declarations).  Running a quick test against RBI with this file confirms that:

```brightscript
Library "v30/bslCore.brs"

sub main()
    print "Hello, world!"
end sub

Library "v30/bslCore.brs"  ' <-- syntax error on line 7
```

Looks like we'll need to replace the depth tracking logic with a simple "have we done anything that's not a library yet?" check :cry:  I'm sorry it took so long to find this!  Somewhere within the `declaration()` function feels reasonable for this, but I haven't tested that at all.

**Comment by sjbarag** [comment]
Thanks for keeping this up to date and dealing with my absence over the past few months, @TwitchBronBron!

File: src/parser/Parser.ts [hunk 2]
```diff
@@ -589,6 +637,50 @@ export class Parser {
...
 
+        function libraryStatement(): Stmt.Library | undefined {
+            let libraryStatement = new Stmt.Library({
+                library: advance(),
+                //grab the next token only if it's a string
+                filePath: peek().kind === Lexeme.String ? advance() : undefined
+            });
+
+            //no token following library keyword token
+            if (!libraryStatement.tokens.filePath && peek().kind === Lexeme.Newline) {
```
[the remark this answers is not in the document]
Comment (TwitchBronBron) [reply src/parser/Parser.ts]: Sure, I'll change it. And I put together an example to show you how we should not support colon as a separator between library calls...but then found out that's actually supported! 

```
Library "v30/bslCore.brs" : Library "v30/bslCore.brs"

sub main()
    print "Hello, world!"
end sub
```

File: src/parser/Parser.ts (hunk 1 above)
Comment (TwitchBronBron) [reply src/parser/Parser.ts]: I already handle this scenario [here](https://github.com/sjbarag/brs/pull/216/files#diff-0af3c0ebba8c6d54ca06e99fa694be4fR666) 

```
//libraries must be at the very top of the file before any other declarations.
let isAtTopOfFile = true;
for (let statement of statements) {
    //if we found a non-library statement, this statement is not at the top of the file
    if (!(statement instanceof Stmt.Library)) {
        isAtTopOfFile = false;
    }
}
```

File: src/parser/Parser.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (TwitchBronBron) [reply src/parser/Parser.ts]: Fixed

File: src/parser/Parser.ts [hunk 3]
```diff
@@ -589,6 +637,50 @@ export class Parser {
             return new Stmt.ExitFor({ exitFor: keyword });
         }
 
+        function libraryStatement(): Stmt.Library | undefined {
+            let libraryStatement = new Stmt.Library({
+                library: advance(),
+                //grab the next token only if it's a string
+                filePath: peek().kind === Lexeme.String ? advance() : undefined
```
[the remark this answers is not in the document]
Comment (TwitchBronBron) [reply src/parser/Parser.ts]: Fixed

**Comment by TwitchBronBron** [comment]
@sjbarag this is ready for your review again.

File: src/parser/Parser.ts (hunk 1 above)
Comment (sjbarag) [reply src/parser/Parser.ts]: Derp, I missed that!  Thanks for pointing it out :smiley:

**Pull request merged**